engine/src/main/java/org/terasology/rendering/dag/states/State.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag.states;
+
+/**
+ *
+ */
+public interface State {
+    void set(StateValue stateValue);
+
+    StateValue getValue();
```

Comment:
Due to previous comments and architectural changes I think the type returned by getValue() should be dependent on the actual opengl state attribute being affected. So, a state change inheriting from AbstractStageChange could return a float, another an integer, another something else. I'm not sure how to implement this the best. Perhaps the method could return a generic object to be cast appropriately?